.github/workflows/continuous-integration.yml

```
@@ -54,50 +88,67 @@ jobs:
         path: rrtmgp-data
         ref: v1.8.1
     #
-    # Synchronize the package index
-    #
-    - name: Synchronize the package index
-      run: sudo apt-get update
-    #
-    # Install NetCDF-Fortran (compatible with all compilers)
-    #
-    - name: Install NetCDF-Fortran
-      run: sudo apt-get install libnetcdff-dev
-    #
     # Cache Conda packages
     #
     - name: Cache Conda packages
       uses: actions/cache@v4
       with:
         path: ~/conda_pkgs_dir
-        key: conda-pkgs
+        key:
+          ${{ runner.os }}-conda-${{ hashFiles('environment-ci.yml') }}
     #
     # Set up Conda
     #
     - name: Set up Conda
       uses: conda-incubator/setup-miniconda@v3
       with:
         miniforge-version: latest
-        activate-environment: rte_rrtmgp_test
-        environment-file: environment-noplots.yml
-        python-version: 3.11
+        activate-environment: rte_rrtmgp_ci
+        environment-file: environment-ci.yml
         auto-activate-base: false
+        use-mamba: true
         # Use the cache properly:
         use-only-tar-bz2: true
     #
+    # Install NetCDF-Fortran
+    #
+    - name: Install NetCDF-Fortran via Conda
+      if: runner.os == 'Linux'
+      run: conda install -y -c conda-forge netcdf-fortran
+    #
+    # Setup Linux environment variables
+    #
+    - name: Setup env variables - Linux
+      if: runner.os == 'Linux'
+      run: |
+        echo FCINCLUDE=-I ${CONDA_PREFIX}/include >> $GITHUB_ENV
+        echo LDFLAGS=-L ${CONDA_PREFIX}/lib >> $GITHUB_ENV
+        echo LD_LIBRARY_PATH=${CONDA_PREFIX}/lib:${LD_LIBRARY_PATH} >> $GITHUB_ENV
+    #
+    # Setup Windows environment variables
+    #
+    - name: Setup env variables - Windows
+      if: runner.os == 'Windows'
+      run: |
+        echo FCINCLUDE="-I/mingw64/include" >> $GITHUB_ENV
+        echo LDFLAGS="-L/mingw64/lib" >> $GITHUB_ENV
+        echo LD_LIBRARY_PATH="/mingw64/lib:/mingw64/bin:${LD_LIBRARY_PATH}" >> $GITHUB_ENV
```

```suggestion
        echo "FCINCLUDE=-I/mingw64/include" >> $GITHUB_ENV
        echo "LDFLAGS=-L/mingw64/lib" >> $GITHUB_ENV
        echo "LD_LIBRARY_PATH=/mingw64/lib:/mingw64/bin:${LD_LIBRARY_PATH}" >> $GITHUB_ENV
```
Is the `LD_LIBRARY_PATH` variable needed here? If so, do we need the `/mingw64/bin` entry?

---

Same comment as before for the styling, happy to change it to match.

> Is the `LD_LIBRARY_PATH` variable needed here?

Yes, MSYS is not in the runner's PATH, see [docs](https://github.com/actions/runner-images/blob/main/images/windows/Windows2022-Readme.md#notes) and we are not setting the `-Wl,-rpath,/bath/to/libs`. We have to set explicitly `LD_LIBRARY_PATH`.

> If so, do we need the `/mingw64/bin` entry?

This is where the compiler's runtime libraries are. I have no idea what compelled the MSYS packagers to place them there, but I am sure there is a reason which I don't know about.

see output from my fork's Action
 https://github.com/gnikit/rte-rrtmgp/actions/runs/9326569389/job/25675276582#step:4:312

> /mingw64/bin/libgfortran-5.dll